pkg/storage/stores/shipper/table_client.go

```diff
@@ -48,7 +48,7 @@ func (b *boltDBShipperTableClient) Stop() {
 }
 
 func (b *boltDBShipperTableClient) DeleteTable(ctx context.Context, name string) error {
-	objects, dirs, err := b.objectClient.List(ctx, name, delimiter)
+	objects, dirs, err := b.objectClient.List(ctx, name+delimiter, delimiter)
```

Comment:
You should add a test for this.

Reply:
A [test](https://github.com/grafana/loki/blob/b489130828e39c8a7e3c94479691ecedd4d00de8/pkg/storage/stores/shipper/table_client_test.go#L54) is there but it uses filesystem object client which works with or without a delimiter, we don't have an easy way to test this with hosted object stores without mocking the client.